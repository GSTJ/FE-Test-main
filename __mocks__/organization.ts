import type {IOrganization} from '../app/screens/organizations';

export const makeOrganizationMock = ({
  name = 'My Non-Profit',
  uri = 'https://www.example.com/logo.png',
  ein = '123456789',
  type = 'Pet',
  description = 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
} = {}): IOrganization => ({
  name,
  uri,
  ein,
  type,
  description,
});
