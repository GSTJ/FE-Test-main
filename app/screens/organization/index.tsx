import {RouteProp, useRoute} from '@react-navigation/native';
import React from 'react';
import {View, StyleSheet, Image, ScrollView} from 'react-native';
import {FlatList} from 'react-native-gesture-handler';
import {Chip, Title, Paragraph} from 'react-native-paper';
import {RootStackParamList, ScreenNames} from '../../types/reactNavigation';

type OrganizationScreenRouteProp = RouteProp<
  RootStackParamList,
  ScreenNames.Organization
>;

const Organization = () => {
  const {params} = useRoute<OrganizationScreenRouteProp>();

  return (
    <ScrollView
      testID="organization-screen"
      style={styles.container}
      bounces={false}>
      <Image
        accessibilityRole="image"
        testID="organization-logo"
        source={{uri: params.uri}}
        style={styles.image}
      />
      <View style={styles.content}>
        <Title accessibilityRole="header">Categories</Title>
        <FlatList /* To showcase a possible use of multiple categories */
          data={[params.type]}
          bounces={false}
          horizontal
          showsHorizontalScrollIndicator={false}
          renderItem={({item: category}) => (
            <Chip accessibilityRole="text" style={styles.chip}>
              {category}
            </Chip>
          )}
          keyExtractor={item => String(item)}
        />
        <Title accessibilityRole="header">Description</Title>
        <Paragraph accessibilityRole="text">{params.description}</Paragraph>
      </View>
    </ScrollView>
  );
};

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#f7f7f7',
  },
  content: {
    padding: 20,
    borderTopWidth: 1,
    borderColor: '#e0e0e0',
  },
  image: {
    height: 300,
  },
  chip: {
    marginRight: 10,
    marginBottom: 10,
  },
});

export default Organization;
