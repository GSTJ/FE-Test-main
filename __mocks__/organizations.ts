import {makeOrganizationMock} from './organization';

export default [
  makeOrganizationMock({type: 'Animal', name: 'Animal Shelter', ein: '1'}),
  makeOrganizationMock({type: 'Animal', name: 'Animal Rescue', ein: '2'}),
  makeOrganizationMock({type: 'Sport', name: 'Soccer Club', ein: '3'}),
  makeOrganizationMock({type: 'Sport', name: 'Basketball Club', ein: '4'}),
  makeOrganizationMock({type: 'Marine', name: 'Marine Rescue', ein: '5'}),
];
