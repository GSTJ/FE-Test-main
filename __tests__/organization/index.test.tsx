import React from 'react';

import {render} from '@testing-library/react-native';
import Organization from '../../app/screens/organization';
import {makeOrganizationMock} from '../../__mocks__/organization';

const mockParams = makeOrganizationMock();

jest.mock('@react-navigation/native', () => ({
  useRoute: () => ({params: mockParams}),
  useNavigation: () => ({
    goBack: jest.fn(),
  }),
}));

describe('Organization screen', () => {
  it('matches snapshot', () => {
    const component = render(<Organization />);
    expect(component).toMatchSnapshot();
  });

  it('renders correct description', () => {
    const component = render(<Organization />);

    expect(component.getByText(mockParams.description)).toBeTruthy();
  });

  it('renders correct category', () => {
    const component = render(<Organization />);

    expect(component.getByText(mockParams.type)).toBeTruthy();
  });

  it('renders correct company logo', () => {
    const component = render(<Organization />);

    const organizationLogo = component.getByTestId('organization-logo');

    expect(organizationLogo.props.source.uri).toBe(mockParams.uri);
  });
});
