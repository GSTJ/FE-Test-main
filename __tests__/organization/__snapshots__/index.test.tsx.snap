// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organization screen matches snapshot 1`] = `
Array [
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "backgroundColor": "#6200ee",
          "elevation": 4,
          "zIndex": 0,
        },
        Object {
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 3,
            "width": 0,
          },
          "shadowOpacity": 0.24,
          "shadowRadius": 4,
        },
      ]
    }
  >
    <View
      collapsable={false}
      nativeID="animatedComponent"
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#6200ee",
          "elevation": 0,
          "flexDirection": "row",
          "height": 56,
          "marginTop": 0,
          "paddingHorizontal": 4,
        }
      }
    >
      <View
        accessibilityLabel="Back"
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        hitSlop={
          Object {
            "bottom": 6,
            "left": 6,
            "right": 6,
            "top": 6,
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "margin": 6,
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 18,
                "height": 36,
                "width": 36,
              },
              undefined,
              undefined,
            ],
          ]
        }
      >
        <View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
                Object {
                  "height": 24,
                  "width": 24,
                },
              ]
            }
          >
            <View
              collapsable={false}
              nativeID="animatedComponent"
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "transform": Array [
                    Object {
                      "rotate": "0deg",
                    },
                  ],
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                    },
                    Object {
                      "height": 24,
                      "transform": Array [
                        Object {
                          "scaleX": 1,
                        },
                      ],
                      "width": 24,
                    },
                  ]
                }
              >
                <Image
                  source={
                    Object {
                      "testUri": "../../react-native-paper/lib/commonjs/assets/back-chevron.png",
                    }
                  }
                  style={
                    Array [
                      Object {
                        "resizeMode": "contain",
                      },
                      Object {
                        "height": 21,
                        "tintColor": "#ffffff",
                        "width": 21,
                      },
                    ]
                  }
                />
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        accessibilityState={
          Object {
            "disabled": true,
          }
        }
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "flex": 1,
              "paddingHorizontal": 12,
            },
            Array [
              Object {
                "marginLeft": 8,
              },
              Object {
                "alignItems": "center",
              },
              undefined,
            ],
          ]
        }
      >
        <Text
          accessibilityRole="header"
          accessibilityTraits="header"
          accessible={true}
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "textAlign": "left",
              },
              Array [
                Object {
                  "color": "#ffffff",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "fontSize": 17,
                },
                undefined,
              ],
            ]
          }
        >
          My Non-Profit
        </Text>
      </View>
      <View
        style={
          Object {
            "width": 48,
          }
        }
      />
    </View>
  </RCTSafeAreaView>,
  <RCTScrollView
    bounces={false}
    style={
      Object {
        "backgroundColor": "#f7f7f7",
      }
    }
  >
    <View>
      <Image
        source={
          Object {
            "uri": "https://www.example.com/logo.png",
          }
        }
        style={
          Object {
            "borderBottomWidth": 1,
            "borderColor": "#e0e0e0",
            "height": 300,
          }
        }
        testID="organization-logo"
      />
      <View
        style={
          Object {
            "padding": 20,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "textAlign": "left",
              },
              Array [
                Object {
                  "textAlign": "left",
                },
                Object {
                  "color": "rgba(0, 0, 0, 0.87)",
                  "fontFamily": "System",
                  "fontWeight": "500",
                  "writingDirection": "ltr",
                },
                Array [
                  Object {
                    "fontSize": 20,
                    "letterSpacing": 0.15,
                    "lineHeight": 30,
                    "marginVertical": 2,
                  },
                  undefined,
                ],
              ],
            ]
          }
        >
          Categories
        </Text>
        <RCTScrollView
          bounces={false}
          collapsable={false}
          data={
            Array [
              "Pet",
            ]
          }
          getItem={[Function]}
          getItemCount={[Function]}
          horizontal={true}
          keyExtractor={[Function]}
          onContentSizeChange={[Function]}
          onGestureHandlerEvent={[Function]}
          onGestureHandlerStateChange={[Function]}
          onLayout={[Function]}
          onMomentumScrollBegin={[Function]}
          onMomentumScrollEnd={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          removeClippedSubviews={false}
          renderItem={[Function]}
          renderScrollComponent={[Function]}
          scrollEventThrottle={50}
          showsHorizontalScrollIndicator={false}
          stickyHeaderIndices={Array []}
          viewabilityConfigCallbackPairs={Array []}
        >
          <View>
            <View
              onLayout={[Function]}
              style={
                Array [
                  Object {
                    "flexDirection": "row",
                  },
                  null,
                ]
              }
            >
              <View
                collapsable={false}
                nativeID="animatedComponent"
                style={
                  Object {
                    "backgroundColor": "#ebebeb",
                    "borderColor": "#ebebeb",
                    "borderRadius": 16,
                    "borderStyle": "solid",
                    "borderWidth": 0.5,
                    "elevation": 0,
                    "flexDirection": "column",
                    "marginBottom": 10,
                    "marginRight": 10,
                  }
                }
              >
                <View
                  accessibilityRole="button"
                  accessibilityState={
                    Object {
                      "disabled": true,
                      "selected": false,
                    }
                  }
                  accessible={true}
                  focusable={false}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Array [
                      Object {
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "borderRadius": 16,
                        },
                        Object {
                          "flexGrow": 1,
                        },
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "flexDirection": "row",
                          "flexGrow": 1,
                          "paddingLeft": 4,
                          "position": "relative",
                        },
                        Object {
                          "paddingRight": 4,
                        },
                      ]
                    }
                  >
                    <Text
                      numberOfLines={1}
                      selectable={false}
                      style={
                        Array [
                          Object {
                            "color": "#000000",
                            "fontFamily": "System",
                            "fontWeight": "400",
                          },
                          Object {
                            "textAlign": "left",
                          },
                          Array [
                            Object {
                              "lineHeight": 24,
                              "marginVertical": 4,
                              "minHeight": 24,
                              "textAlignVertical": "center",
                            },
                            Object {
                              "color": "rgba(0, 0, 0, 0.87)",
                              "fontFamily": "System",
                              "fontWeight": "400",
                              "marginLeft": 8,
                              "marginRight": 8,
                            },
                            undefined,
                          ],
                        ]
                      }
                    >
                      Pet
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RCTScrollView>
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "textAlign": "left",
              },
              Array [
                Object {
                  "textAlign": "left",
                },
                Object {
                  "color": "rgba(0, 0, 0, 0.87)",
                  "fontFamily": "System",
                  "fontWeight": "500",
                  "writingDirection": "ltr",
                },
                Array [
                  Object {
                    "fontSize": 20,
                    "letterSpacing": 0.15,
                    "lineHeight": 30,
                    "marginVertical": 2,
                  },
                  undefined,
                ],
              ],
            ]
          }
        >
          Description
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "textAlign": "left",
              },
              Array [
                Object {
                  "textAlign": "left",
                },
                Object {
                  "color": "rgba(0, 0, 0, 0.87)",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "writingDirection": "ltr",
                },
                Array [
                  Object {
                    "fontSize": 14,
                    "letterSpacing": 0.25,
                    "lineHeight": 20,
                    "marginVertical": 2,
                  },
                  undefined,
                ],
              ],
            ]
          }
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit.
        </Text>
      </View>
    </View>
  </RCTScrollView>,
]
`;
